src/backend/base/langflow/components/vectorstores/AstraDB.py

```diff
@@ -160,7 +201,159 @@ class AstraVectorStoreComponent(LCVectorStoreComponent):
+            ).to_dict()
+
+            new_parameter_3 = SecretStrInput(
+                name="z_03_provider_api_key",
+                display_name="Provider API Key",
+                info="An alternative to the Astra Authentication that passes an API key for the provider with each request to Astra DB. This may be used when Vectorize is configured for the collection, but no corresponding provider secret is stored within Astra's key management system.",
+            ).to_dict()
+
+            new_parameter_4 = DictInput(
+                name="z_04_authentication",
+                display_name="Authentication parameters",
+                is_list=True,
+            ).to_dict()
+
+            self.insert_in_dict(
+                build_config,
+                "provider",
+                {
+                    "z_00_model_name": new_parameter_0,
+                    "z_01_model_parameters": new_parameter_1,
+                    "z_02_api_key_name": new_parameter_2,
+                    "z_03_provider_api_key": new_parameter_3,
+                    "z_04_authentication": new_parameter_4,
+                },
+            )
+
+        return build_config
+
+    @check_cached_vector_store
+    def build_vectorize_options(self, **kwargs):
+        for attribute in [
+            "provider",
+            "z_00_api_key_name",
+            "z_01_model_name",
+            "z_02_authentication",
+            "z_03_provider_api_key",
+            "z_04_model_parameters",
+        ]:
+            if not hasattr(self, attribute):
+                setattr(self, attribute, None)
+
+        # Fetch values from kwargs if any self.* attributes are None
+        provider_value = self.VECTORIZE_PROVIDERS_MAPPING.get(self.provider, [None])[0] or kwargs.get("provider")
+        authentication = {**(self.z_02_authentication or kwargs.get("z_02_authentication", {}))}
+
+        api_key_name = self.z_00_api_key_name or kwargs.get("z_00_api_key_name")
+        provider_key_name = self.z_03_provider_api_key or kwargs.get("z_03_provider_api_key")
+        if provider_key_name:
+            authentication["providerKey"] = provider_key_name
+        if api_key_name:
+            authentication["providerKey"] = api_key_name
+
+        return {
+            # must match astrapy.info.CollectionVectorServiceOptions
+            "collection_vector_service_options": {
+                "provider": provider_value,
+                "modelName": self.z_01_model_name or kwargs.get("z_01_model_name"),
+                "authentication": authentication,
+                "parameters": self.z_04_model_parameters or kwargs.get("z_04_model_parameters", {}),
+            },
+            "collection_embedding_api_key": self.z_03_provider_api_key or kwargs.get("z_03_provider_api_key"),
+        }
+
+    def build_vector_store(self, vectorize_options=None):
```

Comment:
Curious about the parameter addition here. Is it only used for testing purposes? In the main path, there's no scenario where this wouldn't be `None,` right?

Reply:
That's correct! I left a comment up above for a similar reason why it was done, but if there's a better / easier way i missed let me know cuz i didnt like it either. The goal was basically to allow for the tests to execute successfully with pytest (and for what its worth, the 5 test_astra_vectorize tests do in fact work for me) but i would love if rather than having these optional parameters it simulated more like how the UI executes the code. The dynamic inputs is the challenge...

Reply:
Hm yeah it's very possible we don't yet have any tests that use dynamic inputs, and thus haven't had to support a way to do that. I'll pull this and play around a bit this afternoon just to try as well